### DISCUSSION TITLE
including C <math> library

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#586] [Q&A] [answered]
Hi, I want to use `exp` in my main code, but when I compile it using Aideck docker, it says `undefined reference to 'exp'`. I googled it and found that maybe using `-lm` as a flag can solve this problem. But I don't know how should I do this. Should I modify it in `freeRTOS_rules.mk`? Thanks.

**Comment by krichardsson** [comment, marked as answer]
Which CPU are your writing code for?
If you are working with C (and not C++) you should probably use math.h, like this
```C
#include <math.h>
```

there might also be various flavors depending on if you use a float or double
```C 
expf(myFloat);
exp(myDouble);
```

**Reply by noahzn** [reply]
Thank you!

**Comment by ChenyuZhaoTHU** [comment]
Hi! Have you solved the issue? I still do not know how to solve it like using sqrt() or cos() in math.h. Thanks.
